libs/SmartSync/src/com/salesforce/androidsdk/smartsync/util/SoqlSyncDownTarget.java

```diff
@@ -119,17 +128,49 @@ public JSONArray continueFetch(SyncManager syncManager) throws IOException, JSON
         if (nextRecordsUrl == null) {
             return null;
         }
-
         RestRequest request = new RestRequest(RestRequest.RestMethod.GET, nextRecordsUrl, null);
         RestResponse response = syncManager.sendSyncWithSmartSyncUserAgent(request);
         JSONObject responseJson = response.asJSONObject();
         JSONArray records = responseJson.getJSONArray(Constants.RECORDS);
 
-        // Capture next records url
+        // Captures next records URL.
         nextRecordsUrl = JSONObjectHelper.optString(responseJson, Constants.NEXT_RECORDS_URL);
         return records;
     }
 
+    @Override
+    public Set<String> getListOfRemoteIds(SyncManager syncManager, Set<String> localIds) {
+        if (localIds == null) {
+            return null;
+        }
+        final String idFieldName = getIdFieldName();
+        final Set<String> remoteIds = new HashSet<String>();
+
+        // Alters the SOQL query to get only IDs.
+        final StringBuilder soql = new StringBuilder("SELECT ");
+        soql.append(idFieldName);
+        soql.append(" FROM");
+        final String[] fromClause = query.toLowerCase().split("from");
```

Comment:
```SOQL``` doesn't appear to be case-sensitive AFAIK. I tried the query in our sample app both ways on the ```Contact``` object, and all lowercase appears to work fine too.